### PR TITLE
Rename to pymc-extras

### PR DESCRIPTION
**Pull request by fonnesbeck**
Changes instances of `*-experimental` to `*-extras` throughout. Also renames the module itself, and deals with renaming on all imports. Additional cleanup as dictated by linting.

### CODE REVIEW

**Comment by ricardoV94** [comment]
Want to also update imports/docstrings in the codebase?

**Comment by wd60622** [comment]
And the environment names in conda-envs/

**Comment by ricardoV94** [comment]
@OriolAbril anything special needed for RTD?

**Comment by fonnesbeck** [comment]
OK, that took some doing. Should be clean now.

**Comment by wd60622** [comment]
What will the new import look like? 🤔

**Comment by fonnesbeck** [comment]
`import pymc_extras as pmx`

**Comment by fonnesbeck** [comment]
This appears to have broken the `statespace` module quite badly.

**Comment by ricardoV94** [comment]
We can live without it

**Comment by wd60622** [comment]
This have functional changes as well?

File: .github/workflows/pypi.yml [hunk 1]
```diff
@@ -28,18 +28,18 @@ jobs:
             cd test-sdist
             python -m venv venv-sdist
             venv-sdist/bin/python -m pip install numpy
-            venv-sdist/bin/python -m pip install ../dist/pymc_experimental*.tar.gz
+            venv-sdist/bin/python -m pip install ../dist/pymc_extras*.tar.gz
             echo "Checking import and version number (on release)"
-            venv-sdist/bin/python -c "import pymc_experimental as pmx; assert pmx.__version__ == '${{ github.ref_name }}'[1:] if '${{ github.ref_type }}' == 'tag' else True; print(pmx.__version__)"
+            venv-sdist/bin/python -c "import pymc as pmx; assert pmx.__version__ == '${{ github.ref_name }}'[1:] if '${{ github.ref_type }}' == 'tag' else True; print(pmx.__version__)"
```
Comment (ricardoV94) [inline .github/workflows/pypi.yml:33]: This is wrong and elsewhere in this file

**Review by ricardoV94 — CHANGES_REQUESTED** [review]

File: pymc_extras/version.py
Comment (wd60622) [inline pymc_extras/version.py:1]: Should the version number be changed since it's a new package?

Comment (ricardoV94) [reply pymc_extras/version.py:1]: That's fine, we can continue from the old numbering. We did that with pymc and I think it's a bit easier to follow along

File: .github/workflows/pypi.yml (hunk 1 above)
Comment (fonnesbeck) [reply .github/workflows/pypi.yml:33]: Oops.

**Comment by fonnesbeck** [comment]
And just like that, tests pass.

File: pymc_extras/version.py
Comment (fonnesbeck) [reply pymc_extras/version.py:1]: Yes, it will make it obvious when the renaming occurred.

**Comment by fonnesbeck** [comment]
Can this be merged?

File: docs/statespace/filters.rst [hunk 2]
```diff
@@ -2,14 +2,11 @@
...
+.. automodule:: pymc_extras.statespace.filters
 .. autosummary::
    :toctree: generated
 
     StandardFilter
     UnivariateFilter
-    SteadyStateFilter
     KalmanSmoother
-    SingleTimeseriesFilter
-    CholeskyFilter
```
Comment (ricardoV94) [inline docs/statespace/filters.rst:14]: Unrelated changes? Did you merge main instead of rebasing? Makes it a bit hard to check

**Comment by ricardoV94** [comment]
@fonnesbeck I'm going to squash the commits and rebase to see if the github changed lines become more streamlined

File: docs/statespace/filters.rst (hunk 2 above)
Comment (fonnesbeck) [reply docs/statespace/filters.rst:14]: I removed these because they did not seem to exist in the code base, so tests were failing.

**Comment by ricardoV94** [comment]
@fonnesbeck I reverted all accidental functional changes I could find.

**Review by OriolAbril — APPROVED** [review]

**Comment by ricardoV94** [comment]
> Is it ok to keep these urls or do we want those updated?

Whatever is less work for us

File: docs/statespace/filters.rst (hunk 2 above)
Comment (ricardoV94) [reply docs/statespace/filters.rst:14]: The cholesky, was renamed to SquareRootFilter. The others were removed as you said

**Review by ricardoV94 — APPROVED** [review]
Let's see if the tests pass

**Comment by ricardoV94** [comment]
All existing PRs will have to be rebased and we should create a pypi pckage ASAP @fonnesbeck

**Comment by fonnesbeck** [comment]
It doesn't look like we can rename pymc-experimental, so I will have to create a new project. The next version is slated to be 0.1.5 but shall we jump to 0.2.0 for this?

**Comment by ricardoV94** [comment]
> It doesn't look like we can rename pymc-experimental, so I will have to create a new project. The next version is slated to be 0.1.5 but shall we jump to 0.2.0 for this? Or dare we make this a major version?

I would do 0.2. Maybe we'll have something very flashy for 1.0

**Comment by fonnesbeck** [comment]
> I would do 0.2. Maybe we'll have something very flashy for 1.0

Coward.
